apps/financial-aid/api/src/app/modules/auth/auth.controller.ts

```diff
@@ -142,12 +147,19 @@ export class AuthController {
       return res.redirect('/?villa=innskraning-ogild')
     }
 
+    let hasApplied = false
+
+    if (service === 'osk') {
+      hasApplied = await this.authService.checkUserHistory(islandUser.kennitala)
```

Comment:
We should change the logic here and use the power of graphql to fetch this data only for osk users. Do you want to pair on how we do that or should I write it up?